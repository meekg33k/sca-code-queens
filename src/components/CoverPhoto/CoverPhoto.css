.coverPhoto {
    margin: 15px 0 15px 45px;
    text-align: center;
    border-radius: 10px 10px;
    height: 80vh;
    box-shadow: 0px 3px 6px #00000029 ;
  -webkit-box-shadow: 0px 3px 6px #00000029;
  -moz-box-shadow: 0px 3px 6px #00000029;
}

.coverText {
    display: flex;
   display: inline-flex;
   margin: 45px;
   font-size: 18px;
   font-weight: 600;
  
}

.firstText {
    margin-right: 10px;
}

/*.coverPhoto > img {
    margin: 5px 0; 
}*/