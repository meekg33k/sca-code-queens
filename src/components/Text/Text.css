.greenish {
  color: #1abc9c;
}

.greyish {
  color: #7f8c8d;
}

.white {
  color: #ffffff;
}

.extrasmall {
  font-size: 22px;
}

.default,
.small,
.medium,
.large {
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
}

.default {
  font-size: 18px;
}

.small {
  font-size: 28px;
}

.medium {
  font-size: 32px;
}

.large {
  font-size: 42px;
}
