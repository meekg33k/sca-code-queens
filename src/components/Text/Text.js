import React from "react"

// //functional component
//const Text = (props) => {
//    //const {text} = props
//    const text = props.text
//    return (
//        <div>
//            <h1>{text}</h1>
//        </div>
//    )  
//}
//export default Text

//class based component
class Text extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
        text: props.text   //Get initial text from props 
        }
    }
    render() {
        return (
            <div>
               <p>{this.state.text}</p>
            </div>
        )
    }
}

export default Text