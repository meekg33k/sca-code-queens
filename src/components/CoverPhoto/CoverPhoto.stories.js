import React from "react";
import {storiesOf} from "@storybook/react";
import {withInfo} from "@storybook/addon-info";
import {withKnobs, text} from "@storybook/addon-knobs";
import CoverPhoto from "./index";

storiesOf("CoverPhoto", module)
    .addDecorator(
        withInfo({
            text: "This is a CoverPhoto component"
        })
    )

    .addDecorator(withKnobs()
    )

    .add("CoverPhoto", () => (
        <CoverPhoto
            name={text("name", "name of artiste")}
            coverimg={text("coverimg","image of artiste")} 
        />
    ));