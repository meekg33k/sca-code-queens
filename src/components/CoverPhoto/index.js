import React from "react";
import Image from "../Image";
import Text from "../Text"
import "./CoverPhoto.css"


const CoverPhoto = ({name, coverimg}) => {
    return (
        <div className="coverPhoto">
            <div className="coverText">
                <div className="firstText">
                    <Text text="Playing DJ:" color="greyish" size="extrasmall" />
                </div>
            <Text text={name} color="greyish" size="extrasmall" />
            </div>
            <Image source={coverimg} alt="cover photo" width= {200} height= {200} />
            
        </div>
    )

};

export default CoverPhoto;