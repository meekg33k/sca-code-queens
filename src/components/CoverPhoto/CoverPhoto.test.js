import React from "react";
import { mount } from "enzyme";
import Text from "../Text";
import CoverPhoto from "./index";


describe("CoverPhoto Component render", () => {
    it("renders correctly", () => {
        const wrapper = mount(<CoverPhoto name="name of artiste" />)
        expect(wrapper).toMatchSnapshot();
    })

    it("renders image correctly", () => {
        const wrapper = mount(<CoverPhoto coverimg="name of artiste" />)
        expect(wrapper).toMatchSnapshot();
    })

    it("renders text on the page", () => {
        const wrapper = mount(<CoverPhoto />);
        const findText = wrapper.find(Text);
        expect(findText.length).toBe(2);
    });

    it("renders the name of the artiste correctly", () => {
        const wrapper = mount(<CoverPhoto name="The name of the artiste" />);
        expect(
          wrapper
            .find(Text)
            .at(1)
            .props().text
        ).toBe("The name of the artiste");
      });

    it("renders the image of the artiste", () => {
    const wrapper = mount(<CoverPhoto />);
    const artisteImg = wrapper.find(Image);
    expect(artisteImg.length).toBe(0);
    });

})