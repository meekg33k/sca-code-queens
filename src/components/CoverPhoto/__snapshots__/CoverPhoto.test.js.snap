// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoverPhoto Component render renders correctly 1`] = `ReactWrapper {}`;

exports[`CoverPhoto Component render renders image correctly 1`] = `ReactWrapper {}`;
